### PR TITLE
Added MyAvatar.hmdLeanRecenterEnabled property

### PR DESCRIPTION
**Pull request by hyperlogic**
Used to disable the 'room-scale' avatar re-centering code.  Disabling this can prevent sliding when the avatar is supposed to be sitting or mounted on a stationary object.

Also, removed a bunch of old, unused leaning and torso twisting code.

### CODE REVIEW

**Comment by hyperlogic** [comment]
Test plan:
   * enter either the vive or oculus display plugin while wearing an hmd.
   * physically lean backward by moving your chair backward or by taking a step backward.
   * observe that the avatar also takes a step backward to recenter underneath your head.
   * open the script console.
   * type `MyAvatar.hmdLeanRecenterEnabled = true;`
   * physically lean backward by moving your chair backward or by taking a step backward.
   * observe that the avatar no longer takes a step backward to recenter underneath your head.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 9292a9ce0b244579900f66143795ffb4ef29bd2f available at: https://gist.github.com/ab5bcd4ed6a7ed2dc9aa94bdb7028a57<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8010.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8010.dmg

**Comment by thoys** [comment]
## QA report

I did all the steps of your test plan, except I've typed `MyAvatar.hmdLeanRecenterEnabled = false;`,  I think thats what you ment. It functions well as for what it has to do, although the sphere-ui interface doesn't keep following you. Not sure if this is expected, but for what this PR needs to do I give it a pass.

:+1:  QA Successfully passed

**Comment by hifi-gustavo** [comment]
Build console output for commit: 30d8ae36e8db11d17f7e4ea237ed89771630d221 available at: https://gist.github.com/c45b7405ec06b87527cfaf4276bd72c0<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8010.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8010.dmg

**Comment by Triplelexx** [comment]
Tested MyAvatar.characterControllerEnabled property, it's working great.

File: interface/src/avatar/MyAvatar.cpp [hunk 1]
```diff
@@ -1839,12 +1834,16 @@ void MyAvatar::updateMotionBehaviorFromMenu() {
...
-    bool checked = menu->isOptionChecked(MenuOption::EnableCharacterController);
-    if (!ghostingAllowed) {
-        checked = true;
-    }
+    _characterController.setEnabled(ghostingAllowed ? enabled : true);
+}
 
-    _characterController.setEnabled(checked);
+bool MyAvatar::getCharacterControllerEnabled() {
+    if (QThread::currentThread() != thread()) {
```
Comment (ZappoMan) [inline interface/src/avatar/MyAvatar.cpp]: does this really need to be implemented this way? Can't we use an atomic<bool> or something like that?

**Comment by hifi-gustavo** [comment]
Build console output for commit: 1aae22f5a59246e795801aa09ecf07bd93ee50af available at: https://gist.github.com/c45b7405ec06b87527cfaf4276bd72c0<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8010.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8010.dmg

**Pull request merged**